### PR TITLE
Added support for ARM responses without 'value' key

### PR DESCRIPTION
**Pull request by giladsu**

### CODE REVIEW

File: src/SDK/SmartDetectorsShared/Clients/ExtendedAzureResourceManagerClient.cs [hunk 1]
```diff
@@ -336,12 +336,21 @@ public async Task<List<JObject>> ExecuteArmQueryAsync(ResourceIdentifier resourc
                 string responseContent = await response.Content.ReadAsStringAsync();
                 JObject responseObject = JObject.Parse(responseContent);
 
-                var returnedObjects = responseObject["value"].ToObject<List<JObject>>();
-                allItems.AddRange(returnedObjects);
+                string valueKey = "value";
```
Comment (EranGon) [inline src/SDK/SmartDetectorsShared/Clients/ExtendedAzureResourceManagerClient.cs]: >valueKey [](start = 23, length = 8)

make const

File: src/SDK/SmartDetectorsShared/Clients/ExtendedAzureResourceManagerClient.cs [hunk 2]
```diff
@@ -336,12 +336,21 @@ public async Task<List<JObject>> ExecuteArmQueryAsync(ResourceIdentifier resourc
...
+                }
+                else
+                {
+                    allItems.Add(responseObject);
+                }
 
                 // Link to next page
                 string nextLinkToken = responseObject.GetValue("nextLink", StringComparison.InvariantCulture)?.ToString();
-                nextLink = (string.IsNullOrWhiteSpace(nextLinkToken) || !returnedObjects.Any()) ? null : new Uri(nextLinkToken);
+                nextLink = (string.IsNullOrWhiteSpace(nextLinkToken) || !(returnedObjects?.Any()).GetValueOrDefault() ? null : new Uri(nextLinkToken);
```
Comment (EranGon) [inline src/SDK/SmartDetectorsShared/Clients/ExtendedAzureResourceManagerClient.cs]: >GetValueOrDefault [](start = 98, length = 17)

please put "?? false" inside the braces - it's clearer

File: src/SDK/SmartDetectorsShared/Clients/ExtendedAzureResourceManagerClient.cs (hunk 1 above)
Comment (giladsu) [reply src/SDK/SmartDetectorsShared/Clients/ExtendedAzureResourceManagerClient.cs]: True dat.

File: src/SDK/SmartDetectorsShared/Clients/ExtendedAzureResourceManagerClient.cs (hunk 2 above)
Comment (giladsu) [reply src/SDK/SmartDetectorsShared/Clients/ExtendedAzureResourceManagerClient.cs]: Agreed. Done.

**Pull request merged**